docs/dev/adding_instruments.rst

```diff
@@ -1157,21 +1157,48 @@ Device tests
 It can be useful as well to test the code against an actual device. The necessary device setup instructions (for example: connect a probe to the test output) should be written in the header of the test file or test methods. There should be the connection configuration (for example serial port), too.
 In order to distinguish the test module from protocol tests, the filename should be :code:`test_instrumentName_with_device.py`, if the device is called :code:`instrumentName`.
 
-Mark tests that require instrument hardware to be `skipped <https://docs.pytest.org/en/stable/how-to/skipping.html>`_ by default.
-If the whole test module requires hardware, add this at module level/after the import statements:
+To make it easier for others to run these tests using their own instruments, we recommend to use :code:`pytest.fixture` to create an instance of the instrument class.
+It is important to use the specific argument name :code:`connected_device_address` and define the scope of the fixture to only establish a single connection to the device.
+This ensures two things:
+First, it makes it possible to specify the address of the device to be used for the test using the :code:`--device-address` command line argument.
+Second, tests using this fixture, i.e. tests that rely on a device to be connected to the computer are skipped by default when running pytest.
+This is done to avoid that the tests are run on a CI server that does not have the device connected.
```

Comment:
```suggestion
This is done to avoid that the tests are run on that does not have the device connected.
```
This is more generally valid than just for CI.

Reply:
I agree with the idea, but not the actual suggestion.

Reply:
oops.

Reply:
changed in [6e2abc1](https://github.com/pymeasure/pymeasure/pull/789/commits/6e2abc1be8ebe1045fdd4602a8813f658ad13ea9)

Reply:
If you approve the change, @bilderbuchi , you might merge this PR.

Reply:
Sure, thanks!